### PR TITLE
Resource Updates

### PR DESCRIPTION
**Pull request by damoodamoo**
# PR for issue #1243 

## What it does
Feeds the newly `PATCH`ed properties through to the runner, along with the `upgrade` command. A large majority of the existing deployment / status update code remains as-is, with just a few tweaks to better handle upgrades. The `PATCH` endpoints now return an operation doc + location header which can be used to track the upgrade.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -284,14 +284,18 @@ static-web-upload:
 	&& ./templates/core/terraform/scripts/upload_static_web.sh
 
 test-e2e-smoke:
-	$(call target_title, "Running E2E smoke tests") && \
-	cd e2e_tests && \
-	python -m pytest -m smoke --verify $${IS_API_SECURED:-true} --junit-xml pytest_e2e_smoke.xml
+	$(call target_title, "Running E2E smoke tests") \
+	&& cp --force ./templates/core/.env ./e2e_tests/.env \
+	&& . ./devops/scripts/load_env.sh ./templates/core/private.env \
```
Comment (ross-p-smith) [inline Makefile:289]: private.env will not be there when run in the GitHub Build

**Comment by damoodamoo** [comment]
bad build + deletion stuck, re-pr-ing

**Pull request closed without merging**